org.lflang/src/org/lflang/federated/generator/FedGenerator.java

```diff
@@ -94,10 +94,14 @@ public synchronized void report(int id, int x, Procedure1<Integer> callback) {
     private final Map<Integer, FederateInstance> federateByID = new LinkedHashMap<>();
     /**
      * The federation RTI properties, which defaults to 'localhost: 15045'.
+     * The "overridden" Pair guarantees following user directives, without setting
+     * the default values to be "null". A particular use case is when the lf file
+     * sets RTI properties, and the command line overrides these.
      */
     final LinkedHashMap<String, Object> federationRTIProperties = CollectionLiterals.newLinkedHashMap(
         Pair.of("host", "localhost"),
-        Pair.of("port", 0) // Indicator to use the default port, typically 15045.
+        Pair.of("port", 0), // Indicator to use the default port, typically 15045.
+        Pair.of("overridden", false)
```

Comment:
Perhaps this should be named "host_overridden"? It isn't clear which property it is referring to.